lib/cloud/aws/policy.go

```diff
@@ -213,7 +213,8 @@ func NewPolicies(accountID string, iamClient iamiface.IAMAPI) Policies {
 // * `iam:DeletePolicyVersion`: wildcard ("*") or policy that will be created;
 // * `iam:CreatePolicyVersion`: wildcard ("*") or policy that will be created;
 func (p *policies) Upsert(ctx context.Context, policy *Policy) (string, error) {
-	policyARN := fmt.Sprintf("arn:aws:iam::%s:policy/%s", p.accountID, policy.Name)
+	partitionId := p.iamClient.(*iam.IAM).PartitionID
```

Comment:
I have a feeling this will break unit test. I suggest adding `parititionID` to `policies` struct and `NewPolicies` constructor.

Reply:
You are right, I will improve this code